src/operator/contrib/all_finite.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file all_finite.cc 
+ * \brief operator for checking if a group of array is all finite
+ * \author Clement Fuji Tsang
+ */
+#include "./all_finite-inl.h"
+
+namespace mxnet {
+namespace op {
+
+template<typename DType>
+struct AllFiniteCPUKernel {
+  MSHADOW_XINLINE static void Map(int i, const DType* in, float* out) {
+    bool is_finite = true;
+    is_finite = in[i] == in[i] ? is_finite : false;
+    if (!is_finite) {
+      out[0] = 0.;
+    }
+  }
+};
+
+inline void AllFiniteCPU(const nnvm::NodeAttrs& attrs,
+                         const OpContext &ctx,
+                         const std::vector<TBlob> &inputs,
+                         const std::vector<OpReqType> &req,
+                         const std::vector<TBlob> &outputs) {
+  using namespace mxnet_op;
+  Stream<cpu>* s = ctx.get_stream<cpu>();
+  const AllFiniteParam& op_param = nnvm::get<AllFiniteParam>(attrs.parsed);
+  Tensor<cpu, 2, float> out = outputs[0].FlatTo2D<cpu, float>(s);
+  if (op_param.init_output)
+    out = 1.;
+  MSHADOW_REAL_TYPE_SWITCH(inputs[0].type_flag_, DType, {
```

Comment:
nit: indentation

Reply:
I don't see any problem, could you be more specific here?